### PR TITLE
Slim 1382 mbus encryption key status

### PR DESCRIPTION
**Pull request by smvdheijden**

### CODE REVIEW

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/services/ActionMapperService.java [hunk 1]
```diff
@@ -193,27 +203,32 @@ private void postConstruct() {
...
 
             if (customValueToDtoConverter != null) {
                 actionValueObjectDtoList.add(new ActionDto(customValueToDtoConverter.convert(action, smartMeter)));
             } else {
-
                 final ConfigurableMapper mapper = CLASS_TO_MAPPER_MAP.get(action.getClass());
                 final Class<? extends ActionRequestDto> clazz = CLASS_MAP.get(action.getClass());
                 if (mapper != null) {
+
+                    if (action instanceof MBusActionRequest) {
```
Comment (robindenadel) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/services/ActionMapperService.java]: MBus -> Mbus for all applicable elements

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/services/ConfigurationService.java [hunk 2]
```diff
@@ -764,4 +765,47 @@ public void handleConfigureDefinableLoadProfileResponse(final DeviceMessageMetad
                 result, exception, null, deviceMessageMetadata.getMessagePriority()),
                 deviceMessageMetadata.getMessageType());
     }
+
```
Comment (robindenadel) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/services/ConfigurationService.java]: MBus -> Mbus for all applicable elements

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/GetMBusEncryptionKeyStatusResponseMessageProcessor.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/**
```
Comment (robindenadel) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/GetMBusEncryptionKeyStatusResponseMessageProcessor.java]: MBus -> Mbus for all applicable elements

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/ws/messageprocessors/GetMBusEncryptionKeyStatusRequestMessageProcessor.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+/**
```
Comment (robindenadel) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/ws/messageprocessors/GetMBusEncryptionKeyStatusRequestMessageProcessor.java]: MBus -> Mbus for all applicable elements

File: osgp-adapter-domain-smartmetering/src/test/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/GetMBusEncryptionKeyStatusRequestDataMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
+/**
```
Comment (robindenadel) [inline osgp-adapter-domain-smartmetering/src/test/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/GetMBusEncryptionKeyStatusRequestDataMapperTest.java]: MBus -> Mbus for all applicable elements

File: osgp-adapter-domain-smartmetering/src/test/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/GetMBusEncryptionKeyStatusResponseDataMapperTest.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
+/**
```
Comment (robindenadel) [inline osgp-adapter-domain-smartmetering/src/test/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/GetMBusEncryptionKeyStatusResponseDataMapperTest.java]: MBus -> Mbus for all applicable elements

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/services/ConfigurationService.java [hunk 7]
```diff
@@ -385,6 +385,34 @@ public String enqueueSetEncryptionKeyExchangeOnGMeterRequest(
         return correlationUid;
     }
 
+    public String enqueueGetMBusEncryptionKeyStatusRequest(final String organisationIdentification,
```
Comment (robindenadel) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/services/ConfigurationService.java]: MBus -> Mbus for all applicable elements

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/infra/jms/SmartMeteringRequestMessageType.java [hunk 8]
```diff
@@ -46,5 +46,6 @@
     GENERATE_AND_REPLACE_KEYS,
     GET_M_BUS_DEVICE_ON_CHANNEL,
     CONFIGURE_DEFINABLE_LOAD_PROFILE,
-    SET_MBUS_USER_KEY_BY_CHANNEL;
+    SET_MBUS_USER_KEY_BY_CHANNEL,
+    GET_M_BUS_ENCRYPTION_KEY_STATUS;
```
Comment (robindenadel) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/infra/jms/SmartMeteringRequestMessageType.java]: M_BUS -> MBUS

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/infra/jms/messageprocessor/GetMBusEncryptionKeyStatusResponseMessageProcessor.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
+/**
```
Comment (robindenadel) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/infra/jms/messageprocessor/GetMBusEncryptionKeyStatusResponseMessageProcessor.java]: MBus -> Mbus for all applicable elements

File: osgp-adapter-ws-smartmetering/src/test/java/com/alliander/osgp/adapter/ws/smartmetering/application/mapping/GetMBusEncryptionKeyStatusRequestMappingTest.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
+/**
```
Comment (robindenadel) [inline osgp-adapter-ws-smartmetering/src/test/java/com/alliander/osgp/adapter/ws/smartmetering/application/mapping/GetMBusEncryptionKeyStatusRequestMappingTest.java]: MBus -> Mbus for all applicable elements

File: osgp-adapter-ws-smartmetering/src/test/java/com/alliander/osgp/adapter/ws/smartmetering/application/mapping/GetMBusEncryptionKeyStatusResponseMappingTest.java [hunk 11]
```diff
@@ -0,0 +1,46 @@
+/**
```
Comment (robindenadel) [inline osgp-adapter-ws-smartmetering/src/test/java/com/alliander/osgp/adapter/ws/smartmetering/application/mapping/GetMBusEncryptionKeyStatusResponseMappingTest.java]: MBus -> Mbus for all applicable elements

File: osgp-core/src/main/resources/db/migration/V20171106150600666__Adds_device_function_mapping_GetMBusEncryptionKeyStatus.sql [hunk 12]
```diff
@@ -0,0 +1,10 @@
+DO
+$$
+BEGIN
+
+IF NOT EXISTS (SELECT 1 FROM device_function_mapping WHERE "function" = 'GET_M_BUS_ENCRYPTION_KEY_STATUS' AND function_group = 'OWNER') THEN
+	INSERT INTO device_function_mapping (function_group, "function") VALUES ('OWNER', 'GET_M_BUS_ENCRYPTION_KEY_STATUS');
+END IF;
+
```
Comment (robindenadel) [inline osgp-core/src/main/resources/db/migration/V20171106150600666__Adds_device_function_mapping_GetMBusEncryptionKeyStatus.sql]: M_BUS -> MBUS

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/DeviceFunction.java [hunk 13]
```diff
@@ -85,5 +85,6 @@
     GET_LIGHT_SENSOR_STATUS,
     SET_DEVICE_LIFECYCLE_STATUS,
     CONFIGURE_DEFINABLE_LOAD_PROFILE,
-    SET_MBUS_USER_KEY_BY_CHANNEL;
+    SET_MBUS_USER_KEY_BY_CHANNEL,
+    GET_M_BUS_ENCRYPTION_KEY_STATUS;
```
Comment (robindenadel) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/DeviceFunction.java]: M_BUS -> MBUS

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/GetMBusEncryptionKeyStatusRequestData.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
+/**
```
Comment (robindenadel) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/GetMBusEncryptionKeyStatusRequestData.java]: MBus -> Mbus for all applicable elements

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/GetMBusEncryptionKeyStatusResponseData.java [hunk 15]
```diff
@@ -0,0 +1,33 @@
+/**
```
Comment (robindenadel) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/GetMBusEncryptionKeyStatusResponseData.java]: MBus -> Mbus for all applicable elements

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/MBusActionRequest.java [hunk 16]
```diff
@@ -0,0 +1,33 @@
+/**
```
Comment (robindenadel) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/MBusActionRequest.java]: MBus -> Mbus for all applicable elements

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/smartmetering/exceptions/GatewayDeviceInvalidForMBusDeviceException.java [hunk 17]
```diff
@@ -0,0 +1,24 @@
+/**
```
Comment (robindenadel) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/smartmetering/exceptions/GatewayDeviceInvalidForMBusDeviceException.java]: MBus -> Mbus for all applicable elements

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/smartmetering/exceptions/GatewayDeviceNotSetForMBusDeviceException.java [hunk 18]
```diff
@@ -0,0 +1,24 @@
+/**
```
Comment (robindenadel) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/smartmetering/exceptions/GatewayDeviceNotSetForMBusDeviceException.java]: MBus -> Mbus for all applicable elements

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/EncryptionKeyStatusType.java [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+
+public enum EncryptionKeyStatusType {
+
+    NO_ENCRYPTION_KEY(0),
+    ENCRYPTION_KEY_SET(1),
+    ENCRYPTION_KEY_TRANSFERED(2),
+    ENCRYPTION_KEY_SET_AND_TRANSFERRED(3),
+    ENCRYPTION_KEY_IN_USE(4);
+
+    private int value;
```
Comment (bvdzwet) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/EncryptionKeyStatusType.java]: See comments about the protocol detail specific value used with `EncryptionKeyStatusTypeDto` in the Shared repository.

**Pull request merged**